CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ### Breaking changes
 
 * `RealmResults.distinct()` returns a new `RealmResults` object instead of filtering on the original object (#2947).
+* `RealmResults` is auto-updated all the time. Any transaction on the caller thread which may have impact on the order or elements of the `RealmResults` will change the `RealmResults` immediately instead of change it in the next event loop. Iterator behavior of `RealmResults` stays the same, transaction inside the iterating still works as expected.
```

Comment:
Does that mean that even the good ol' `for(int i = results.size()-1; i >= 0; i--)` won't work either?